api/tools/src/main/java/org/pmiops/workbench/tools/ManageClusters.java

```diff
@@ -91,6 +101,50 @@ private static void listClusters(String apiUrl) throws IOException, ApiException
     System.out.println(String.format("listed %d clusters", count.get()));
   }
 
+  private static void describeCluster(
+      String apiUrl, String workbenchProjectId, String workbenchServiceAccount, String clusterId)
+      throws IOException, ApiException {
+    String[] parts = clusterId.split("/");
+    if (parts.length != 2) {
+      System.err.println(
+          String.format(
+              "given cluster ID '%s' is invalid, wanted format 'project/clusterName'", clusterId));
+      return;
+    }
+    String clusterProject = parts[0];
+    String clusterName = parts[1];
+
+    // Leo's getCluster API swagger tends to be outdated; issue a raw getCluster request to ensure
+    // we get all available information for debugging.
+    ClusterApi client = newApiClient(apiUrl);
+    com.squareup.okhttp.Call call =
+        client.getClusterCall(
+            clusterProject,
+            clusterName,
+            /* progressListener */ null,
+            /* progressRequestListener */ null);
+    ApiResponse<Object> resp = client.getApiClient().execute(call, Object.class);
+
+    // Parse the response as well so we can log specific structured fields.
+    Cluster cluster = PRETTY_GSON.fromJson(PRETTY_GSON.toJson(resp.getData()), Cluster.class);
+
+    System.out.println(PRETTY_GSON.toJson(resp.getData()));
+    System.out.printf("\n\nTo inspect logs in cloud storage, run the following:\n\n");
+
+    // TODO(PD-4740): Use impersonation here instead.
+    String keyPath = String.format("/tmp/%s-key.json", workbenchProjectId);
+    System.out.printf(
+        "    gcloud iam service-accounts keys create %s --iam-account %s\n",
+        keyPath, workbenchServiceAccount);
+    System.out.printf("    gcloud auth activate-service-account --key-file %s\n\n", keyPath);
+    System.out.printf("    gsutil ls gs://%s/**\n", cluster.getStagingBucket());
+    System.out.printf("    gsutil cat ... # inspect or copy logs\n\n");
+    System.out.printf("    # Delete the key when done\n");
```

Comment:
If I'm [reading this right](https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts.keys), you can set an explicit expiration time by passing `validBeforeTime`, but I think it's only for the REST API and not the gcloud command.